platforms/access-bridge/src/errors.ts

```diff
@@ -1,196 +1,132 @@
-import { ServerResponse } from 'node:http';
-
+import { Response } from 'express';
 import { JWError, JWErrorResponse } from '@jwp/ott-common/types/errors.js';
 
-import { RequestMethod } from './http.js';
-
-/**
- * List of possible error codes based on RFC-JW03-Delivery API
- * https://www.notion.so/jwplayer/RFC-JW03-Delivery-API-fc35c5f5d300445eba4004de970d06b4
- */
-export enum ErrorCode {
-  BadRequestError = 'bad_request',
-  ParameterMissing = 'parameter_missing',
-  ParameterInvalid = 'parameter_invalid',
-  Unauthorized = 'unauthorized',
-  Forbidden = 'forbidden',
-  NotFound = 'not_found',
-  MethodNotAllowed = 'method_not_allowed',
-  InternalError = 'internal_error',
+// Define types for error codes and status codes
+export type ErrorCode = keyof typeof ErrorDefinitions;
+export type ErrorStatusCode = (typeof ErrorDefinitions)[ErrorCode]['statusCode'];
+
+// Define context types for each error
+interface BaseContext {
+  description?: string;
 }
 
-const ErrorCodeDescription: Record<ErrorCode, string> = {
-  [ErrorCode.BadRequestError]: 'The request was not constructed correctly.',
-  [ErrorCode.ParameterMissing]: 'Required parameter {value} is missing.',
-  [ErrorCode.ParameterInvalid]: 'Parameter {value} is invalid. {reason}.',
-  [ErrorCode.Unauthorized]: 'Missing or invalid auth credentials.',
-  [ErrorCode.Forbidden]: 'Access to the requested resource is not allowed.',
-  [ErrorCode.NotFound]: 'The requested resource could not be found.',
-  [ErrorCode.MethodNotAllowed]: 'The requested resource only supports {value} requests.',
-  [ErrorCode.InternalError]: 'An error was encountered while processing the request. Please try again.',
-};
-
-/**
- * Allowed response status codes.
- */
-export type ErrorStatusCode = 400 | 401 | 403 | 404 | 405 | 500;
-
-/**
- * Base class for errors.
- */
-export abstract class AccessBridgeError {
-  private readonly descriptionOverride: string | null;
-
-  protected constructor(description?: string) {
-    this.descriptionOverride = description || null;
-  }
+interface ParameterMissingContext extends BaseContext {
+  parameterName: string;
+}
 
-  abstract get code(): ErrorCode;
-  abstract get statusCode(): ErrorStatusCode;
-  abstract get headers(): Iterable<[string, string]>;
-  protected abstract get defaultDescription(): string;
+interface ParameterInvalidContext extends BaseContext {
+  parameterName: string;
+  reason?: string;
+}
 
-  get description(): string {
-    return this.descriptionOverride || this.defaultDescription;
+// Unified Error Definitions with details and creation functions
+export const ErrorDefinitions = {
```

Comment:
Did you consider making these all classes that inherit from `AccessBirdgeError`? The `create` prop looks like it could be a constructor.